pennylane/qchem/molecule.py

```diff
@@ -126,16 +127,34 @@ def __init__(
         if l is None:
             l = [i[0] for i in self.basis_data]
 
+        if len(
+            set(
+                qml.math.get_interface(x)
+                for x in [coordinates, alpha, coeff]
+                if qml.math.get_interface(x) != "numpy"
+            )
+        ) > 1 and (alpha is not None or coeff is not None):
+            warnings.warn(
+                "The parameters coordinates, coeff, and alpha are not of the same interface. Please use the same interface for all 3 or there may be unintended behavior.",
```

Comment:
You are allowed to mix numpy with any of the other interfaces. So using numpy + jax is considered "the same interface" but using pnp + jax is considered heresy.

Reply:
I honestly don't like it as I feel it might come to bite us back while searching for the jit compatibility. If the interface is `jax`, I would want everything to remain within `jax` wherever possible (and same thing for `numpy`).